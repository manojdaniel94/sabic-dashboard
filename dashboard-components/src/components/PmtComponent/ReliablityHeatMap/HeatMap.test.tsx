import React from "react";
import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import ReliabilityHeatMap from "./ReliablityHeatMap";

describe("ReliabilityHeatMap component", () => {
  const statusData = {
    assetOff: { value: 1 },
    warning: { value: 2 },
    normal: { value: 3 },
    assetUnderRisk: { value: 4 },
  };

  const toolTipData = [
    { assetId: "1", assetHealthIndex: 50 },
    { assetId: "2", assetHealthIndex: 80 },
    { assetId: "3", assetHealthIndex: 60 },
    { assetId: "4", assetHealthIndex: 90 },
  ];

  const selectedHeatStatus = "";
  const setMouseHover = jest.fn();
  const mouseHover = { over: false, statusName: "" };

  it("renders the component", () => {
    const { container } = render(
      <ReliabilityHeatMap
        statusData={statusData}
        toolTipData={toolTipData}
        selectedHeatStatus={selectedHeatStatus}
        setMouseHover={setMouseHover}
        mouseHover={mouseHover}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it("displays asset-off data", () => {
    const { getByText } = render(
      <ReliabilityHeatMap
        statusData={statusData}
        toolTipData={toolTipData}
        selectedHeatStatus="Asset-off"
        setMouseHover={setMouseHover}
        mouseHover={mouseHover}
      />
    );

    expect(getByText("1")).toBeInTheDocument();
    expect(getByText("Asset-Off")).toBeInTheDocument();
    expect(getByText("Off")).toBeInTheDocument();
    expect(getByText("1")).toHaveClass("hi-off");
  });

  it("displays normal data", () => {
    const { getByText } = render(
      <ReliabilityHeatMap
        statusData={statusData}
        toolTipData={toolTipData}
        selectedHeatStatus="Normal"
        setMouseHover={setMouseHover}
        mouseHover={mouseHover}
      />
    );

    expect(getByText("2")).toBeInTheDocument();
    expect(getByText("Normal")).toBeInTheDocument();
    expect(getByText("Health Index > 70%")).toBeInTheDocument();
    expect(getByText("3")).toHaveClass("hi-normal");
  });

  it("displays warning data", () => {
    const { getByText } = render(
      <ReliabilityHeatMap
        statusData={statusData}
        toolTipData={toolTipData}
        selectedHeatStatus="Warning"
        setMouseHover={setMouseHover}
        mouseHover={mouseHover}
      />
    );

    expect(getByText("3")).toBeInTheDocument();
    expect(getByText("Warning")).toBeInTheDocument();
    expect(getByText("Health Index < 70%")).toBeInTheDocument();
    expect(getByText("2")).toHaveClass("hi-warning");
  });

  it("displays asset under risk data", () => {
    const { getByText } = render(
      <ReliabilityHeatMap
        statusData={statusData}
        toolTipData={toolTipData}
        selectedHeatStatus="Asset Under Risk"
        setMouseHover={setMouseHover}
        mouseHover={mouseHover}
      />
    );

    // expect(getByText("4")).toBeInTheDocument();
    expect(getByText("Asset Under Risk")).toBeInTheDocument();
    expect(getByText("Health Index Under Risk")).toBeInTheDocument();
    expect(getByText("4")).toHaveClass("hi-risk");
  });
});
