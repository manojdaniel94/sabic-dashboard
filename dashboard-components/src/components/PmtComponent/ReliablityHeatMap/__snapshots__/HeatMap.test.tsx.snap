// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReliabilityHeatMap component renders the component 1`] = `
<div>
  <div
    class="pmt-health-index"
  >
    <div
      class="pmt-heatmap"
    >
      <img
        src="image_Heatmap.svg"
      />
    </div>
    <div
      class="pmt-heatmapvalues"
    >
      <div
        class="hi-value1"
      >
        <span
          class="hi-numbers hi-off"
        >
          <div
            class="hi-popup hi-pos-off hidden"
          >
            <h3>
              HEALTH INDEX
               
              <a
                class="hi-pop-close"
                href="#"
              >
                 
                X
              </a>
            </h3>
            <div
              class="hi-popbox"
            >
              <div
                class="hi-popuprow"
              >
                <span>
                  1
                </span>
                <span>
                  <i />
                  50
                  %
                </span>
              </div>
              <div
                class="hi-popuprow"
              >
                <span>
                  2
                </span>
                <span>
                  <i />
                  80
                  %
                </span>
              </div>
              <div
                class="hi-popuprow"
              >
                <span>
                  3
                </span>
                <span>
                  <i />
                  60
                  %
                </span>
              </div>
              <div
                class="hi-popuprow"
              >
                <span>
                  4
                </span>
                <span>
                  <i />
                  90
                  %
                </span>
              </div>
            </div>
          </div>
          1
        </span>
        <span
          class="hi-name"
        >
          Asset-Off
        </span>
        <span
          class="hi-status"
        >
          Off
        </span>
      </div>
      <div
        class="hi-value2"
      >
        <span
          class="hi-numbers hi-normal"
        >
          <div
            class="hi-popup hi-pos-normal hidden"
          >
            <h3>
              HEALTH INDEX
               
              <a
                class="hi-pop-close"
                href="#"
              >
                 
                X
              </a>
            </h3>
            <div
              class="hi-popbox"
            >
              <div
                class="hi-popuprow"
              >
                <span>
                  1
                </span>
                <span>
                  <i />
                  50
                  %
                </span>
              </div>
              <div
                class="hi-popuprow"
              >
                <span>
                  2
                </span>
                <span>
                  <i />
                  80
                  %
                </span>
              </div>
              <div
                class="hi-popuprow"
              >
                <span>
                  3
                </span>
                <span>
                  <i />
                  60
                  %
                </span>
              </div>
              <div
                class="hi-popuprow"
              >
                <span>
                  4
                </span>
                <span>
                  <i />
                  90
                  %
                </span>
              </div>
            </div>
          </div>
           
          2
        </span>
        <span
          class="hi-name"
        >
          Normal
        </span>
        <span
          class="hi-status"
        >
          Health Index &gt; 70%
        </span>
      </div>
      <div
        class="hi-value3"
      >
        <span
          class="hi-numbers hi-warning"
        >
          <div
            class="hi-popup hi-pos-warning hidden"
          >
            <h3>
              HEALTH INDEX
               
              <a
                class="hi-pop-close"
                href="#"
              >
                 
                X
              </a>
            </h3>
            <div
              class="hi-popbox"
            >
              <div
                class="hi-popuprow"
              >
                <span>
                  1
                </span>
                <span>
                  <i />
                  50
                  %
                </span>
              </div>
              <div
                class="hi-popuprow"
              >
                <span>
                  2
                </span>
                <span>
                  <i />
                  80
                  %
                </span>
              </div>
              <div
                class="hi-popuprow"
              >
                <span>
                  3
                </span>
                <span>
                  <i />
                  60
                  %
                </span>
              </div>
              <div
                class="hi-popuprow"
              >
                <span>
                  4
                </span>
                <span>
                  <i />
                  90
                  %
                </span>
              </div>
            </div>
          </div>
           
          3
        </span>
        <span
          class="hi-name"
        >
          Warning
        </span>
        <span
          class="hi-status"
        >
          Health Index &lt; 70%
        </span>
      </div>
      <div
        class="hi-value4"
      >
        <span
          class="hi-numbers hi-risk"
        >
          <div
            class="hi-popup hi-pos-risk hidden"
          >
            <h3>
              HEALTH INDEX
               
              <a
                class="hi-pop-close"
                href="#"
              >
                 
                X
              </a>
            </h3>
            <div
              class="hi-popbox"
            >
              <div
                class="hi-popuprow"
              >
                <span>
                  1
                </span>
                <span>
                  <i />
                  50
                  %
                </span>
              </div>
              <div
                class="hi-popuprow"
              >
                <span>
                  2
                </span>
                <span>
                  <i />
                  80
                  %
                </span>
              </div>
              <div
                class="hi-popuprow"
              >
                <span>
                  3
                </span>
                <span>
                  <i />
                  60
                  %
                </span>
              </div>
              <div
                class="hi-popuprow"
              >
                <span>
                  4
                </span>
                <span>
                  <i />
                  90
                  %
                </span>
              </div>
            </div>
          </div>
          4
        </span>
        <span
          class="hi-name"
        >
          Asset Under Risk
        </span>
        <span
          class="hi-status"
        >
          Health Index Under Risk
        </span>
      </div>
    </div>
  </div>
</div>
`;
